rocketchat.go

```
@@ -101,22 +86,22 @@ func SendNotification(rtClient RocketChatClient, data template.Data, config Conf
 	}
 
 	if channelName == "" {
```

The function signature indicates we are returning an error, however as of now we are only printing the fact the channel wasn't found and we return nil in the end. I don't know if the missing channel name is worth an error, but in case it is I would simply return the error and remove the else clause:

```go
if channelName == "" {
    return t.Error("Exception: Channel name not found. Please specify a default_channel_name in the configuration.")
}

channelID, errRoom := rtClient.GetChannelId(channelName)
if errRoom != nil {
    log.Errorf("Error to get room ID: %v", errRoom)
	return errRoom
}
...
```

---

@rolandvarga The problem here is that the calling webhook function is not sophisticated enough, and will consider this error as an authentication error... 
But you are right that it should be like you said, but the calling webhook must be changed accordingly: you can try the change in another issue/PR!

---

@louisfelix I see the portion you're referring to just now. It makes sense now to not return the error!